django/contrib/admin/views/main.py

```diff
@@ -312,6 +312,9 @@ def get_results(self, request):
         result_count = paginator.count
 
         # Get the total number of objects, with no admin filters applied.
+        # Note this isn't necessarily the same as result_count in the case of
+        # no filtering. Filters defined in list_filters may still apply some
+        # default filtering which may be removed with query parameters.
```

Comment:
Do you think future readers could benefit from an explicit: "The apparent query duplication when counting results can not be simplified since...."?

Reply:
Not really I feel this is clear & concise explanation that they're not (necessarily) equal without elaboration, which is how comments should be imho 😋

Reply:
Fair enough, thanks!